From d8bf28e9ec1a5d854c3fa6f17001db2e8bc59a7a Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Mon, 9 Oct 2017 14:23:18 +0200
Subject: [PATCH 08/69] qemu-kvm-ma: define only pseries-rhel7.5.0 machine type
 for -ma

RH-Author: Laurent Vivier <lvivier@redhat.com>
Message-id: <20171009142318.6262-5-lvivier@redhat.com>
Patchwork-id: 77032
O-Subject: [RHV7.5 qemu-kvm-rhev PATCH 4/4] qemu-kvm-ma: define only pseries-rhel7.5.0 machine type for -ma
Bugzilla: 1478478
RH-Acked-by: David Gibson <dgibson@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

qemu-kvm-ma doesn't support pseries-rhel7.2.0,
pseries-rhel7.3.0, pseries-rhel7.4.0 whereas
qemu-kvm-rhev does. And as the sources are shared
we use a configuration switch to disable this part
of the code.

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/ppc/spapr.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index cb317f8..3c598fd 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -3734,6 +3734,7 @@ DEFINE_SPAPR_MACHINE(2_8, "2.8", false);
     },
 #endif
 
+#if defined(CONFIG_RHV)
 static void phb_placement_2_7(sPAPRMachineState *spapr, uint32_t index,
                               uint64_t *buid, hwaddr *pio,
                               hwaddr *mmio32, hwaddr *mmio64,
@@ -3782,6 +3783,7 @@ static void phb_placement_2_7(sPAPRMachineState *spapr, uint32_t index,
      * window into contiguous 32-bit and 64-bit windows
      */
 }
+#endif /* CONFIG_RHV */
 
 #if 0 /* Disabled for Red Hat Enterprise Linux */
 static void spapr_machine_2_7_instance_options(MachineState *machine)
@@ -3960,6 +3962,8 @@ static void spapr_machine_rhel750_class_options(MachineClass *mc)
 
 DEFINE_SPAPR_MACHINE(rhel750, "rhel7.5.0", true);
 
+#if defined(CONFIG_RHV)
+
 /*
  * pseries-rhel7.4.0
  * like SPAPR_COMPAT_2_9
@@ -4079,6 +4083,7 @@ static void spapr_machine_rhel720_class_options(MachineClass *mc)
 }
 
 DEFINE_SPAPR_MACHINE(rhel720, "rhel7.2.0", false);
+#endif /* CONFIG_RHV */
 
 static void spapr_machine_register_types(void)
 {
-- 
1.8.3.1

